### PR TITLE
backend: Bulk interface for assign conferences

### PR DESCRIPTION
**Pull request by pazembrz**
#1768

### CODE REVIEW

File: backend/inspirehep/assign/tasks.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+def assign_paper_to_conference(literature_recids, conference_recid):
+    try:
+        conference = ConferencesRecord.get_record_by_pid_value(conference_recid)
+    except PIDDoesNotExistError:
+        LOGGER.exception(
+            "Cannot assign papers to conference. Conference does not exist.",
+            conference_recid=conference_recid,
+        )
+    conference_ref = conference.get("self")
+    cnum = conference.get("cnum")
```
Comment (MJedr) [inline backend/inspirehep/assign/tasks.py:53]: Aren't they always present?

File: backend/inspirehep/assign/tasks.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+
+def assign_conference(record, conference_ref, cnum):
+    builder = LiteratureBuilder(record=record)
+    if not cnum:
+        raise MissingArgumentError("cnum is required.")
+    if not conference_ref:
+        raise MissingArgumentError("$ref is required.")
+
+    if not {"proceedings", "conference paper"}.intersection(
```
Comment (MJedr) [inline backend/inspirehep/assign/tasks.py:93]: Why is that? For example if `document_type` is `article` why you are changing it?

**Review by MJedr — COMMENTED** [review]
So finally we always pass recids to the task (even if all the records were selected)?

File: backend/inspirehep/assign/tasks.py (hunk 1 above)
Comment (pazembrz) [reply backend/inspirehep/assign/tasks.py:53]: https://github.com/inspirehep/inspire-schemas/blob/afc2553e150fc83e72aa3187926caa126a0a096d/inspire_schemas/records/conferences.yml#L98-L100 https://github.com/inspirehep/inspire-schemas/blob/afc2553e150fc83e72aa3187926caa126a0a096d/inspire_schemas/records/conferences.yml#L265-L266
They are not mandatory. It's possible that some old conference is missing those.

File: backend/inspirehep/assign/tasks.py (hunk 2 above)
Comment (pazembrz) [reply backend/inspirehep/assign/tasks.py:93]: > So finally we always pass recids to the task (even if all the records were selected)?
Yes.

Comment (pazembrz) [reply backend/inspirehep/assign/tasks.py:93]: I'm not changing this, I'm adding another document_type. This requirement was added when I was working on the task https://github.com/inspirehep/inspirehep/issues/1768#issuecomment-800209835

**Comment by pazembrz** [comment]
> So finally we always pass recids to the task (even if all the records were selected)?

Yes.

File: backend/inspirehep/assign/views.py [hunk 3]
```diff
@@ -64,9 +66,9 @@ def assign_to_author(from_author_recid, to_author_recid, literature_recids):
     unstub_author_by_recid(to_author_recid)
 
 
-@blueprint.route("", methods=["POST"])
+@blueprint.route("authors", methods=["POST"])
 @login_required_with_roles([Roles.cataloger.value])
-def assign_view():
+def author_assign_view():
```
Comment (drjova) [inline backend/inspirehep/assign/views.py:71]:
```suggestion
def authors_assign_view():
```

File: backend/inspirehep/assign/views.py [hunk 4]
```diff
@@ -82,3 +84,37 @@ def assign_view():
...
+
+
+@blueprint.route("literature_conferences", methods=["POST"])
+@login_required_with_roles([Roles.cataloger.value])
+def literature_assign_conferences_view():
+    body = request.get_json()
+    conference_recid = body.get("conference_recid")
+    literature_recids = body.get("literature_recids")
+
+    if not conference_recid or not literature_recids:
```
Comment (drjova) [inline backend/inspirehep/assign/views.py:96]: Instead I would use `@parser.use_args` decorator to check the args

File: backend/inspirehep/assign/views.py [hunk 5]
```diff
@@ -82,3 +84,37 @@ def assign_view():
     if to_author_recid is None:
         return jsonify({"stub_author_id": stub_author_id}), 200
     return jsonify({"message": "Success"}), 200
+
+
+@blueprint.route("literature_conferences", methods=["POST"])
```
Comment (drjova) [inline backend/inspirehep/assign/views.py:89]:
```suggestion
@blueprint.route("conference", methods=["POST"])
```

File: backend/inspirehep/assign/views.py [hunk 6]
```diff
@@ -64,9 +66,9 @@ def assign_to_author(from_author_recid, to_author_recid, literature_recids):
     unstub_author_by_recid(to_author_recid)
 
 
-@blueprint.route("", methods=["POST"])
+@blueprint.route("authors", methods=["POST"])
```
Comment (drjova) [inline backend/inspirehep/assign/views.py:69]:
```suggestion
@blueprint.route("author", methods=["POST"])
```

File: backend/inspirehep/assign/views.py (hunk 3 above)
Comment (pazembrz) [reply backend/inspirehep/assign/views.py:71]: But this is not bulk like action, we are assigning/replacing one author for another in multiple papers. I don't think that `authors` here will be correct.

File: backend/inspirehep/assign/views.py (hunk 6 above)
Comment (pazembrz) [reply backend/inspirehep/assign/views.py:69]: :+1:

File: backend/inspirehep/assign/views.py (hunk 5 above)
Comment (pazembrz) [reply backend/inspirehep/assign/views.py:89]: :+1:

File: backend/inspirehep/assign/views.py (hunk 4 above)
Comment (pazembrz) [reply backend/inspirehep/assign/views.py:105]: :+1:

File: backend/inspirehep/assign/tasks.py [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    retry_kwargs={"max_retries": 6},
+    autoretry_for=(
+        NoResultFound,
+        StaleDataError,
+        DisconnectionError,
+        TimeoutError,
+        UnboundExecutionError,
+        ResourceClosedError,
+        OperationalError,
+    ),
```
Comment (drjova) [inline backend/inspirehep/assign/tasks.py:42]: aren't we use this errors everywhere? can we move them and reuse the same?

File: backend/inspirehep/assign/views.py [hunk 8]
```diff
@@ -82,3 +87,43 @@ def assign_view():
...
+            400,
+        )
+
+    if not isinstance(literature_recids, list):
+        return (
+            jsonify(
+                {"message": "literature_recids has to be a list of literature recids"}
+            ),
+            400,
+        )
```
Comment (drjova) [inline backend/inspirehep/assign/views.py:121]: we don't need these checks with the parser

Comment (pazembrz) [reply backend/inspirehep/assign/views.py:121]: Ah, yes. You are right ;)

File: backend/inspirehep/assign/tasks.py (hunk 7 above)
Comment (pazembrz) [reply backend/inspirehep/assign/tasks.py:42]: We use similar ones in migrator tasks, they are not 100% identical, so I would have to create a common group of exceptions and manually add rest of them in proper places. It's not used that often to be worth additional complexity IMHO.

**Review by drjova — APPROVED** [review]

**Review by drjova — APPROVED** [review]

**Pull request merged**